make/autoconf/jdk-version.m4

```diff
@@ -69,34 +69,38 @@ AC_DEFUN_ONCE([JDKVER_SETUP_JDK_VERSION_NUMBERS],
   AC_SUBST(JDK_RC_PLATFORM_NAME)
   AC_SUBST(HOTSPOT_VM_DISTRO)
 
+  # Note: UTIL_ARG_WITH treats empty strings as valid values when OPTIONAL is false!
+  # That said, it allows for using the RESULT of an option as its own DEFAULT, which
+  # is pretty convenient...
+
+  # Outer [ ] to quote m4.
+  [ USERNAME=`$ECHO "$USER" | $TR -d -c '[a-z][A-Z][0-9]'` ]
+
   # Setup username (for use in adhoc version strings etc)
-  AC_ARG_WITH([build-user], [AS_HELP_STRING([--with-build-user],
-      [build username to use in version strings])])
-  if test "x$with_build_user" = xyes || test "x$with_build_user" = xno; then
-    AC_MSG_ERROR([--with-build-user must have a value])
-  elif test "x$with_build_user" != x; then
-    USERNAME="$with_build_user"
-  else
-    # Outer [ ] to quote m4.
-    [ USERNAME=`$ECHO "$USER" | $TR -d -c '[a-z][A-Z][0-9]'` ]
-  fi
+  UTIL_ARG_WITH(NAME: build-user, TYPE: string,
+    RESULT: USERNAME,
+    DEFAULT: $USERNAME,
```

Comment:
This looks like the same values flow in and out of the function. Maybe call the fallback value `DEFAULT_USERNAME`?

Reply:
Resolved, thanks